### PR TITLE
Issue #3441126: Language configuration create wrong to Original Language at group-type

### PR DESCRIPTION
**Pull request by vcsvinicius**
## Problem
Some installation has a record at config table to original language and it's wrong, because original language will get data from the content and other language should have a record for they data.
When this record exists, the core never get the right data, because config by translation has priorioty to be selected.

## Solution
Create a hook-update to remove all extra record.

## Theme issue tracker
[PROD-27808](https://getopensocial.atlassian.net/browse/PROD-27808)
[#3441126](https://www.drupal.org/project/social/issues/3441126)

## How to test
That error isn't happen for new installation, because probable it was a bug already fixed by community, so to reproduce we need to add a record directly to the database.

- [ ] Install social language
- [ ] Add a second language
- [ ] Go to Group Type configuration
- [ ] Add translate for the groups
- [ ] Go to the table config and look for the translate for each group 'group.type.group_id', change the group id, example: closed_group, flexible_group...
- [ ] Duplicate the record for second language and change the collection column to the original-language
- [ ] With that the code start to get that record created steo before instead of content

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes
This will fix the bug related to group invite, where the user invited as mentioned as invitor.

## Change Record
N/A

## Translations
N/A

[PROD-27808]: https://getopensocial.atlassian.net/browse/PROD-27808?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

File: modules/custom/social_language/social_language.install [hunk 1]
```diff
@@ -64,3 +67,20 @@ function social_language_update_12201(): string {
...
+/**
+ * Removing wrong language configuration for original-language from groups.
+ */
+function social_language_update_12301(): void {
+  $group_types = GroupType::loadMultiple();
+  foreach ($group_types as $group_type) {
+    $configuration_name = sprintf('group.type.%s', $group_type->id());
+    $collection_value = sprintf('language.%s', $group_type->language()->getId());
+
+    Database::getConnection()
```
Comment (robertragas) [inline modules/custom/social_language/social_language.install:80]: Would it maybe be possible to replace the delete query with the following? As it's more specific than a delete query.

    ```
  $config_translation = \Drupal::languageManager()->getLanguageConfigOverride($group_type->language()->getId(), $configuration_name);
      $config_translation->delete();
```

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Review by robertragas — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by BiaInacio** [comment]
Test done, tested with help of @vcsvinicius!
image

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3866-tnndmco8zgrwtyddmrfdjb7kaufqxaut.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/661d608dd5dd4604b78b58ca

**Comment by vcsvinicius** [comment]
O cherry-picked to 12.3.x and 12.4.x branch

**Pull request merged**